### PR TITLE
converting experimental to prototype

### PR DESCRIPTION
**Pull request by parmeet**
This PR changes torchtext's experimental features to prototype features (in-line with other domain libraries)

### CODE REVIEW

**Review by Nayef211 — APPROVED** [review]
LGTM. Should we call out the fact that this change is BC breaking? I know the experimental/prototype folders aren't a part of the release notes so maybe this isn't necessary?

**Review by mthrok — COMMENTED** [review]
The change looks good, but I recommend to keep those APIs in experimental available for a while, by aliasing them.

If some users rely on code from main branch, then from their point of view, the submodule is removed all sudden, without communication. They have to search for the resolution.

Since Python 3.7, one can define module-level `__get_attr__`, in which you can issuer warning
- https://peps.python.org/pep-0562/ 
- https://twitter.com/raymondh/status/1094686528440168453?lang=en

Example in audio

https://github.com/pytorch/audio/blob/main/torchaudio/prototype/ctc_decoder/__init__.py

**Comment by parmeet** [comment]
> The change looks good, but I recommend to keep those APIs in experimental available for a while, by aliasing them.

Thanks @mthrok  for the suggestion. I made the changes to incorporate deprecation warnings. Could you please take a look if this is done right?

File: torchtext/experimental/transforms.py [hunk 1]
```diff
@@ -1,483 +1,61 @@
...
-        # a. generate the masked input tokens
-        # (1) the [MASK] token 80% of the time
-        masked_tokens = self._mask_input(tokens, mask_mask, self.mask_idx)
-        # (2) a random token 10% of the time
-        masked_tokens = self._mask_input(
-            masked_tokens,
-            rand_mask,
-            torch.randint_like(tokens, high=self.vocab_len),
+        warnings.warn(
+            "experimental package has been moved to prototype. You may change all imports from `experimental` to `prototype`",
```
Comment (mthrok) [inline torchtext/experimental/transforms.py:20]: Just to be even less confusing

```suggestion
            "experimental package has been moved to prototype. You may change all imports from `torchtext.experimental` to `torchtext.prototype`",
```

File: torchtext/experimental/transforms.py [hunk 2]
```diff
@@ -1,483 +1,61 @@
...
+            DeprecationWarning,
```
Comment (mthrok) [inline torchtext/experimental/transforms.py:21]:
```suggestion
            UserWarning,
```

`DeprecationWarning` is default not-shown, so I think `UserWarning` is better. (yes, I know DeprecationWarning is used in the example I cited, which I should fix. :) )

File: torchtext/experimental/vectors.py [hunk 3]
```diff
@@ -1,478 +1,34 @@
...
-    }
-
-    file_name = "glove.{}.{}d.txt".format(name, str(dim))
-    if file_name not in valid_glove_file_names:
-        raise ValueError(
-            "Could not find GloVe file with name {}. Please check that `name` and `dim`"
-            "are valid.".format(str(file_name))
+        warnings.warn(
+            "experimental package has been moved to prototype. You may change all imports from `experimental` to `prototype`",
+            DeprecationWarning,
```
Comment (mthrok) [inline torchtext/experimental/vectors.py:10]:
```suggestion
            UserWarning,
```

File: torchtext/experimental/vectors.py [hunk 4]
```diff
@@ -1,478 +1,34 @@
...
-        "glove.6B.300d.txt",
-    }
-
-    file_name = "glove.{}.{}d.txt".format(name, str(dim))
-    if file_name not in valid_glove_file_names:
-        raise ValueError(
-            "Could not find GloVe file with name {}. Please check that `name` and `dim`"
-            "are valid.".format(str(file_name))
+        warnings.warn(
+            "experimental package has been moved to prototype. You may change all imports from `experimental` to `prototype`",
```
Comment (mthrok) [inline torchtext/experimental/vectors.py:9]:
```suggestion
            "experimental package has been moved to prototype. You may change all imports from `torchtext.experimental` to `torchtext.prototype`",
```

File: torchtext/experimental/vocab_factory.py [hunk 5]
```diff
@@ -1,79 +1,14 @@
...
-from torchtext.vocab import Vocab
+        from torchtext.prototype.vocab_factory import build_vocab_from_text_file
 
-__all__ = [
-    "build_vocab_from_text_file",
-    "load_vocab_from_file",
-]
+        warnings.warn(
+            "experimental package has been moved to prototype. You may change all imports from `experimental` to `prototype`",
+            DeprecationWarning,
```
Comment (mthrok) [inline torchtext/experimental/vocab_factory.py:9]:
```suggestion
            UserWarning,
```

File: torchtext/experimental/vocab_factory.py [hunk 6]
```diff
@@ -1,79 +1,14 @@
...
-)
-from torchtext.vocab import Vocab
+        from torchtext.prototype.vocab_factory import build_vocab_from_text_file
 
-__all__ = [
-    "build_vocab_from_text_file",
-    "load_vocab_from_file",
-]
+        warnings.warn(
+            "experimental package has been moved to prototype. You may change all imports from `experimental` to `prototype`",
```
Comment (mthrok) [inline torchtext/experimental/vocab_factory.py:8]:
```suggestion
            "experimental package has been moved to prototype. You may change all imports from `torchtext.experimental` to `torchtext.prototype`",
```

**Review by mthrok — APPROVED** [review]

**Pull request merged**